pkg/provisioners/gcppubsub/dispatcher/receiver/receiver.go

```diff
@@ -34,35 +38,18 @@ type Receiver struct {
 	client client.Client
 
 	pubSubClientCreator util.PubSubClientCreator
-
-	// Note that for all the default* parameters below, these must be kept in lock-step with the
-	// GCP PubSub Dispatcher's reconciler.
-	// Eventually, individual Channels should be allowed to specify different projects and secrets,
-	// but for now all Channels use the same project and secret.
-
-	// defaultGcpProject is the GCP project ID where PubSub Topics and Subscriptions are created.
-	defaultGcpProject string
-	// defaultSecret and defaultSecretKey are the K8s Secret and key in that secret that contain a
-	// JSON format GCP service account token, see
-	// https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-gcloud
-	defaultSecret    *v1.ObjectReference
-	defaultSecretKey string
 }
 
 // New creates a new Receiver and its associated MessageReceiver. The caller is responsible for
 // Start()ing the returned MessageReceiver.
-func New(logger *zap.Logger, client client.Client, pubSubClientCreator util.PubSubClientCreator, defaultGcpProject string, defaultSecret *v1.ObjectReference, defaultSecretKey string) (*Receiver, *provisioners.MessageReceiver) {
+func New(logger *zap.Logger, client client.Client, pubSubClientCreator util.PubSubClientCreator) (*Receiver, []manager.Runnable) {
 	r := &Receiver{
 		logger: logger,
 		client: client,
 
 		pubSubClientCreator: pubSubClientCreator,
-
-		defaultGcpProject: defaultGcpProject,
-		defaultSecret:     defaultSecret,
-		defaultSecretKey:  defaultSecretKey,
 	}
-	return r, r.newMessageReceiver()
+	return r, []manager.Runnable{r.newMessageReceiver()}
```

Comment:
Why does this return an array of runnables? Seems like it's always a single element.

Reply:
Added a comment about how this intended to be used in the future.